add ipex readme (#595)
huggingface/optimum-intel: README.md

# Quick tour

## IPEX
Here is the example of how to use IPEX optimized model to generate texts.
### generate
```diff
import torch
from transformers import AutoTokenizer
- from transformers import AutoModelForCausalLM
+ from optimum.intel.ipex import IPEXModelForCausalLM

> - model = AutoModelForCausalLM.from_pretrained(
> + model = IPEXModelForCausalLM.from_pretrained(
>   "gpt2",
>   torch_dtype=torch.bfloat16,
>   export=True,
> )

Reviewer: ```suggestion

  model_id = "gpt2"
- model = AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.bfloat16)
+ model = IPEXModelForCausalLM.from_pretrained(model_id, torch_dtype=torch.bfloat16, export=True)
```

# Quick tour

## IPEX
Below are examples of how to use IPEX model to generate texts.
### generate
```diff
import torch
from transformers import AutoTokenizer, AutoConfig
- from transformers import AutoModelForCausalLM
+ from optimum.intel.ipex import IPEXModelForCausalLM

config = AutoConfig.from_pretrained("gpt2")

> model = IPEXModelForCausalLM.from_pretrained(

Reviewer: ```suggestion
- model = AutoModelForCausalLM.from_pretrained(
+ model = IPEXModelForCausalLM.from_pretrained(
```

>   "gpt2",

Reviewer: do you have a quantized model under the intel organization https://huggingface.co/Intel that we can directly load instead ?

Author: I am afraid not, but I could upload a quantized one. 
Inputting a common model_id will be more acceptable for users because they might want to try different models with the API. Would like to hear your opinion : )

Reviewer: I'm not sure I agree here, it would be nice to load an IPEX quantized model here to promote this feature support. If we don't do it we should then explain the gain in using IPEX vs torch : what optimization are applied depending on whether we epxort the model to torchscript or not.

Also could you move this section after the OpenVINO section so that we keep the same structure ?

# Quick tour

> 

Reviewer: ```suggestion
```

You can find more examples in the [documentation](https://huggingface.co/docs/optimum/intel/index).

## IPEX
To load your IPEX model, you can just replace your `AutoModelForXxx` class with the corresponding `IPEXModelForXxx` class. You can set `export=True` to load a PyTorch checkpoint, export your model via TorchScript and apply IPEX optimizations : both operators optimization (replaced with customized IPEX operators) and graph-level optimization (like operators fusion) will be applied on your model.
```diff
  from transformers import AutoTokenizer, pipeline
- from transformers import AutoModelForCausalLM

> + from optimum.intel.ipex import IPEXModelForCausalLM

Reviewer: ```suggestion
+ from optimum.intel import IPEXModelForCausalLM
```

You can find more examples in the [documentation](https://huggingface.co/docs/optimum/intel/index).

## IPEX

> With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. It will also apply linear fusion and [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache) for generation. Finally, jit.trace will be applied to change the model to graph mode.
> Here is the example of how to use IPEX optimized model to generate texts.

Reviewer: ```suggestion
To load your IPEX model, you can just replace your `AutoModelForXxx` class with the corresponding `IPEXModelForXxx` class. You can set `export=True` to load a PyTorch checkpoint, export your model via TorchScript and apply IPEX optimizations : both operators optimization (replaced with customized IPEX operators) and graph-level optimization (like operators fusion) will be applied on your model.
```

You can find more examples in the [documentation](https://huggingface.co/docs/optimum/intel/index).

## IPEX
With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. It will also apply linear fusion and [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache) for generation. Finally, jit.trace will be applied to change the model to graph mode.
Here is the example of how to use IPEX optimized model to generate texts.

> ### generate

Reviewer: ```suggestion

```

You can find more examples in the [documentation](https://huggingface.co/docs/optimum/intel/index).

## IPEX
With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. It will also apply linear fusion and [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache) for generation. Finally, jit.trace will be applied to change the model to graph mode.
Here is the example of how to use IPEX optimized model to generate texts.
### generate
```diff

>   import torch

Reviewer: ```suggestion
```

You can find more examples in the [documentation](https://huggingface.co/docs/optimum/intel/index).

## IPEX

> With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. It will also apply linear fusioin and [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache) for generation. Finally, jit.trace will be applied to change the model to graph mode.

Reviewer: ```suggestion
With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. It will also apply linear fusion and [IAKV](https://intel.github.io/intel-extension-for-pytorch/cpu/latest/tutorials/llm.html#indirect-access-kv-cache) for generation. Finally, jit.trace will be applied to change the model to graph mode.
```

Reviewer: > With `export=True`, IPEX model will replace torch linear to ipex linear which prepacks the weights. 

Isn't it also the case for `export=False` ?

Author: The Ipex optimization should be combined with jit.trace. Otherwise, there are no big differences compared to stock PyTorch.
